src/_gettsim/transfers/arbeitsl_geld_2/kindergelduebertrag.py

```diff
@@ -0,0 +1,153 @@
+"""Module for the calculation of the Kindergeldübertrag."""
+
+import numpy as np
+
+from _gettsim.shared import join_numpy, policy_info
+
+aggregate_by_p_id_kindergeldübertrag = {
+    "kindergeldübertrag_m": {
+        "p_id_to_aggregate_by": "p_id_kindergeld_empf",
+        "source_col": "_diff_kindergeld_kindbedarf_m",
+        "aggr": "sum",
+    },
+}
+
+
+@policy_info(end_date="2022-12-31", name_in_dag="_mean_kindergeld_per_child_m")
+def _mean_kindergeld_per_child_gestaffelt_m(
+    kindergeld_m: float,
+    kindergeld_anz_ansprüche: int,
+) -> float:
+    """Kindergeld per child.
+
+    Returns the average Kindergeld per child. If there are no children, the function
+    returns 0. Helper function for `kindergeld_zur_bedarfsdeckung_m`.
+
+    Parameters
+    ----------
+    kindergeld_m
+        See :func:`kindergeld_m`.
+    kindergeld_anz_ansprüche
+        See :func:`kindergeld_anz_ansprüche`.
+
+    Returns
+    -------
+
+    """
+    if kindergeld_anz_ansprüche > 0:
+        out = kindergeld_m / kindergeld_anz_ansprüche
+    else:
+        out = 0.0
+    return out
+
+
+@policy_info(start_date="2023-01-01", name_in_dag="_mean_kindergeld_per_child_m")
+def _mean_kindergeld_per_child_ohne_staffelung_m(
+    kindergeld_params: dict,
+    kindergeld_anz_ansprüche: int,
+) -> float:
+    """Kindergeld per child.
+
+    Returns the (average) Kindergeld per child. Helper function for
+    `kindergeld_zur_bedarfsdeckung_m`.
+
+    Parameters
+    ----------
+    kindergeld_params
+        See params documentation :ref:`kindergeld_params <kindergeld_params>`.
+    alter
+        See basic input variable :ref:`alter`.
+
+    Returns
+    -------
+
+    """
+    return kindergeld_params["kindergeld"] if kindergeld_anz_ansprüche > 0 else 0.0
+
+
+@policy_info(skip_vectorization=True)
+def kindergeld_zur_bedarfsdeckung_m(
+    _mean_kindergeld_per_child_m: float,
+    p_id_kindergeld_empf: np.ndarray[int],
+    p_id: np.ndarray[int],
+) -> float:
```

Comment:
Returns an array of floats, no?

Reply:
I get the following error in the `_convert_data_to_correct_types` when using `numpy.ndarray[float]`:

```text
ValueError: The data types of the following columns are invalid:
E
E            - kindergeld_zur_bedarfsdeckung_m: The internal type numpy.ndarray[float] is not yet supported.
E
E           Note that conversion from floating point to integers or Booleans inherently suffers from approximation error. It might well be that your data seemingly obey the restrictions when scrolling through them, but in fact they do not (for example, because 1e-15 is displayed as 0.0).
E           The best solution is to convert all columns to the expected data types yourself.
```

Other functions in the code use `numpy.ndarray[bool]` as expected return and this works. Any idea @lars-reimann?

Reply:
Looks like GETTSIM raises this error itself. Generally, one should never use floating point numbers to represent exact monetary value, for the reasons cited in the error message. But if a close approximation is fine, we could adjust the error.

Reply:
The problem is that we check for the type of an input Series in `convert_series_to_internal_type`, which needs to be `float`, `bool` etc.:

```
 277             internal_type = None                                                                                                                                            
 278             if column_name in TYPES_INPUT_VARIABLES:                                                                                                                        
 279                 internal_type = TYPES_INPUT_VARIABLES[column_name]                                                                                                          
 280             elif (                                                                                                                                                          
 281                 column_name in functions_overridden                                                                                                                         
 282                 and "return" in functions_overridden[column_name].__annotations__                                                                                           
 283             ):                                                                                                                                                              
 284                 internal_type = functions_overridden[column_name].__annotations__["return"]                                                                                 
 285                                                                                                                                                                             
 286             # Make conversion if necessary                                                                                                                                  
 287             if internal_type and not check_series_has_expected_type(series, internal_type):                                                                                 
 288                 try:                                                                                                                                                        
 289                     data[column_name] = convert_series_to_internal_type(                                                                                                    
 290                         series, internal_type                                                                                                                               
 291                     )                                                                                                                                                       
 292                     collected_conversions.append(                                                                                                                           
 293                         f" - {column_name} from {series.dtype} "                                                                                                            
 294                         f"to {internal_type.__name__}"                                                                                                                      
 295                     )                                                                                                                                                       
 296                                                                                                                                                                             
 297                 except ValueError as e:                                                                                                                                     
 298                     breakpoint()                                                                                                                                            
 299  ->                 collected_errors.append(f" - {column_name}: {e}")                                                                                                       
(Pdb++) column_name
'kindergeld_zur_bedarfsdeckung_m'
(Pdb++) internal_type
numpy.ndarray[float]
(Pdb++) check_series_has_expected_type(series, internal_type)
False
(Pdb++) check_series_has_expected_type(series, float)
True
(Pdb++) functions_overridden[column_name].__policy_info__
*** AttributeError: 'function' object has no attribute '__policy_info__'
```

When we use the correct type annotation for a function that is already vectorised, this fails. @lars-reimann, can you think of a sensible solution here? Probably means making a case distinction regarding the definition of `internal_type` on line 284 above, depending on whether the function has the `skip_vectorization` decorator, though I am afraid the last line of my debugging shows we haven't applied it at that point.

> Other functions in the code use `numpy.ndarray[bool]` as expected return and this works.

Most likely we never use them as input columns in tests directly, so `convert_series_to_internal_type` never gets called. Although I have not checked.

@MImmesberger: I think the short-term hack here is to leave the annotation at `float`.

Reply:
The attribute added by the decorator is called `__info__`, not `__policy_info__`. Something like this should allow detecting whether the function is vectorized:

https://github.com/iza-institute-of-labor-economics/gettsim/blob/add6232460ac7addbd7f82ff21f0c3823e6e89a0/src/_gettsim_tests/test_docs.py#L139